File: dart_lang/lib/a_solid/d_interface_segregation/a_pre_refactor/user_manager.dart
import 'package:dart_lang/a_solid/d_interface_segregation/a_pre_refactor/customer.dart';

abstract class UserManager {
  void subscribeToNewProductsAvailability();
  void subscribeToMSNotifications();
  void updateUserProfile(Customer customer);
  void changePassword(Customer customer, String newPassword);
}

File: PHP/Solid Principles/interface_segregation/problem/UserManager.php
<?php

interface UserManager {
    public function subscribeToNewProductsAvailability(Customer $customer): void;
    public function subscribeToMSNotifications(Customer $customer): void;
    public function updateUserProfile(Customer $customer): void;
    public function changePassword(Customer $customer, string $newPassword): void;
}